plugins/inputs/snmp/snmp.go

```diff
@@ -439,20 +442,40 @@ func (t Table) Build(gs snmpConnection, walk bool) (*RTable, error) {
 		ifv := map[string]interface{}{}
 
 		if !walk {
-			// This is used when fetching non-table fields. Fields configured a the top
-			// scope of the plugin.
-			// We fetch the fields directly, and add them to ifv as if the index were an
-			// empty string. This results in all the non-table fields sharing the same
-			// index, and being added on the same row.
-			if pkt, err := gs.Get([]string{oid}); err != nil {
-				return nil, Errorf(err, "performing get on field %s", f.Name)
-			} else if pkt != nil && len(pkt.Variables) > 0 && pkt.Variables[0].Type != gosnmp.NoSuchObject && pkt.Variables[0].Type != gosnmp.NoSuchInstance {
-				ent := pkt.Variables[0]
-				fv, err := fieldConvert(f.Conversion, ent.Value)
+			if f.UseShell {
+				v, err := snmpVersion(version)
+				if err != nil {
+					return nil, err
+				}
+				out, err := execCmd("snmpget", "-v", v, "-c", community, gs.Host(), oid)
 				if err != nil {
-					return nil, Errorf(err, "converting %q (OID %s) for field %s", ent.Value, ent.Name, f.Name)
+					return nil, Errorf(err, "performing get on field %s", f.Name)
```

Comment:
let's append the output cause exec just gives so bland errors like 'exit status 1'